src/game/behaviors/floating_platform.inc.c

```diff
@@ -45,10 +45,7 @@ void floating_platform_act_0(void) {
 
 void bhv_floating_platform_loop(void) {
     o->oHomeY = floating_platform_find_home_y();
-    if (o->oFloatingPlatformIsOnFloor == 0)
-        o->oAction = 0;
-    else
-        o->oAction = 1;
+    o->oAction = (o->oFloatingPlatformIsOnFloor ? 1 : 0);
```

Comment:
either `o->oAction = o->oFloatingPlatformIsOnFloor;`

or if `oFloatingPlatformIsOnFloor` might not equal `1`

```suggestion
    o->oAction = !!o->oFloatingPlatformIsOnFloor;
```